src/lib/server.ts

```diff
@@ -38,12 +37,14 @@ const patchListen = (app: core.Express) => (listen: core.Express['listen']): Pro
     };
 };
 
-export const createServer = (options?: ServerOptions) => {
+type Unicore = core.Express & { listenAsync: PromisedListen };
+
+export const createServer = (options?: ServerOptions): Unicore => {
     const app = express();
     app.startAt = new Date();
     bindSentry(app, options && options.sentry);
-    override(app, 'listen', patchListen(app));
-    return (app as any) as ({ listen: PromisedListen } & Omit<typeof app, 'listen'>);
+    const core = Object.assign(app, { listenAsync: patchListen(app)(app.listen.bind(app)) });
```

Comment:
:bulb: `return` here

Reply:
:question: just question .. has the object assign any reason here? Can we use spread?

Reply:
I'd rather keep the `core` variable, as it is useful for debugging and readability :grimacing: , please :pray: 

Reason using `Object.assign` here instead of the spread `...` is that assign keeps the structure and only assigns properties, whereas spread creates a completely new object.

So when adding props to a more complicated complicated objects (class instances, functions etc., everything that is not POJO) i prefer `Object.assign` over `spread`.

See for your self using REPL for the differences

```js
> { ...(new Error()),  x: 1 }
{ x: 1 }
> Object.assign(new Error(), { x: 1 })
Error
    at repl:1:15
    at Script.runInThisContext (vm.js:116:20)
    at REPLServer.defaultEval (repl.js:404:29)
    at bound (domain.js:419:14)
    at REPLServer.runBound [as eval] (domain.js:432:12)
    at REPLServer.onLine (repl.js:715:10)
    at REPLServer.emit (events.js:215:7)
    at REPLServer.EventEmitter.emit (domain.js:475:20)
    at REPLServer.Interface._onLine (readline.js:316:10)
    at REPLServer.Interface._line (readline.js:693:8) {
  x: 1
}
```

Reply:
aahh i see, thank you :slightly_smiling_face: